server/sonar-web/src/main/js/components/controls/ModalValidationField.tsx

```diff
@@ -20,67 +20,33 @@
 import * as React from 'react';
 import * as classNames from 'classnames';
 
-type ValidateElement = HTMLInputElement | HTMLTextAreaElement;
-
 interface Props {
-  children: (
-    props: {
-      className?: string;
-      onBlur: () => void;
-      onChange: (evt: React.SyntheticEvent<ValidateElement>) => void;
-      onFocus: () => void;
-      value: string;
-    }
-  ) => React.ReactNode;
+  children: (props: { className?: string }) => React.ReactNode;
   description?: string;
+  dirty: boolean;
   error: string | undefined;
   label?: React.ReactNode;
-  onChange: (value: string) => void;
-  value: string;
-}
-
-interface State {
-  editing: boolean;
   touched: boolean;
 }
 
-export default class ModalValidationField extends React.PureComponent<Props, State> {
-  state: State = { editing: false, touched: false };
-
-  handleBlur = () => this.setState({ editing: false });
-
-  handleChange = (evt: React.SyntheticEvent<ValidateElement>) => {
-    if (!this.state.touched) {
-      this.setState({ touched: true });
-    }
-    this.props.onChange(evt.currentTarget.value);
-  };
-
-  handleFocus = () => this.setState({ editing: true });
-
-  render() {
-    const { description, error, value } = this.props;
-    const { editing, touched } = this.state;
-    const isValid = error === undefined;
-    const showError = touched && !editing && error !== undefined;
-    return (
-      <div className="modal-validation-field">
-        {this.props.label}
-        {this.props.children({
-          className: classNames({
-            'has-error': showError,
-            'is-valid': isValid
-          }),
-          onBlur: this.handleBlur,
-          onChange: this.handleChange,
-          onFocus: this.handleFocus,
-          value
-        })}
-        {showError && <i className="text-danger little-spacer-left icon-alert-error" />}
-        {isValid && <i className="text-success little-spacer-left icon-alert-ok" />}
-        {showError && <p className="text-danger">{error}</p>}
-        {description && <div className="modal-field-description">{description}</div>}
-      </div>
-    );
-  }
+export default function ModalValidationField(props: Props) {
+  const { description, dirty, error } = props;
+
+  const isValid = dirty && props.touched && error === undefined;
+  const showError = dirty && props.touched && error !== undefined;
+  return (
+    <div className="modal-validation-field">
+      {props.label}
+      {props.children({
+        className: classNames({
+          'has-error': showError,
+          'is-valid': isValid
+        })
+      })}
+      {showError && <i className="text-danger little-spacer-left icon-alert-error" />}
```

Comment:
we already have icon components  for these icons, please use them (e.g. AlertErrorIcon)